crates/core/src/error.rs

```diff
@@ -175,69 +175,74 @@ pub enum RusticErrorKind {
+    /// [`std::num::ParseIntError`]
+    #[error(transparent)]
     FromParseIntError(#[from] ParseIntError),
-    /// {0}
+    /// Bytesize parser failed: `{0}`
     FromByteSizeParser(String),
     /// --repack-uncompressed makes no sense for v1 repo!
     RepackUncompressedRepoV1,
-    /// datetime out of range: `{0:?}`
+    /// datetime out of range: `{0}`
     FromOutOfRangeError(#[from] OutOfRangeError),
-    /// node type {0:?} not supported by dump
+    /// node type `{0:?}` not supported by dump
     DumpNotSupported(NodeType),
-    /// {0:?}
+    /// [`serde_json::Error`]
+    #[error(transparent)]
     FromJsonError(#[from] serde_json::Error),
-    /// version {0} is not supported. Allowed values: {1:?}
+    /// version `{0}` is not supported. Allowed values: {1:?}
     VersionNotSupported(u32, RangeInclusive<u32>),
-    /// cannot downgrade version from {0} to {1}
+    /// cannot downgrade version from `{0}` to `{1}`
     CannotDowngrade(u32, u32),
-    /// compression level {0} is not supported for repo v1
+    /// compression level `{0}` is not supported for repo v1
     NoCompressionV1Repo(i32),
-    /// compression level {0} is not supported. Allowed values: {1:?}
+    /// compression level `{0}` is not supported. Allowed values: `{1:?}`
     CompressionLevelNotSupported(i32, RangeInclusive<i32>),
-    /// Size is too large: {0}
+    /// Size is too large: `{0}`
     SizeTooLarge(bytesize::ByteSize),
     /// min_packsize_tolerate_percent must be <= 100
     MinPackSizeTolerateWrong,
     /// max_packsize_tolerate_percent must be >= 100 or 0"
     MaxPackSizeTolerateWrong,
-    /// error creating {0:?}: {1:?}
+    /// error creating `{0:?}`: `{1:?}`
     ErrorCreating(PathBuf, Box<RusticError>),
-    /// error collecting information for {0:?}: {1:?}
+    /// error collecting information for `{0:?}`: `{1:?}`
     ErrorCollecting(PathBuf, Box<RusticError>),
-    /// error setting length for {0:?}: {1:?}
+    /// error setting length for `{0:?}`: `{1:?}`
     ErrorSettingLength(PathBuf, Box<RusticError>),
-    /// {0:?}
+    /// [`rayon::ThreadPoolBuildError`]
+    #[error(transparent)]
     FromRayonError(#[from] rayon::ThreadPoolBuildError),
-    /// conversion from integer failed: `{0:?}`
+    /// Conversion from integer failed: `{0:?}`
     ConversionFromIntFailed(TryFromIntError),
-    /// {0} is not allowed on an append-only repository
+    /// Not allowed on an append-only repository: `{0}`
     NotAllowedWithAppendOnly(String),
     /// Specify one of the keep-* options for forget! Please use keep-none to keep no snapshot.
     NoKeepOption,
-    /// {0:?}
+    /// [`shell_words::ParseError`]
+    #[error(transparent)]
     FromParseError(#[from] shell_words::ParseError),
 }
 
 /// [`CryptoErrorKind`] describes the errors that can happen while dealing with Cryptographic functions
 #[derive(Error, Debug, Display, Copy, Clone)]
 pub enum CryptoErrorKind {
-    /// data decryption failed
+    /// data decryption failed: `{0:?}`
```

Comment:
The underlying error was not shown to the user, so this ended in a simple: `data decryption failed`. Which is not really helpful. The hope is, that `aead:Error` is a bit more helpful here.